src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -146,7 +152,8 @@ public void testFromXContent() throws IOException {
     @Test
     public void testToQuery() throws IOException {
         QueryParseContext context = new QueryParseContext(index, queryParserService);
-        assertLuceneQuery(this.testQuery, this.testQuery.toQuery(context));
+        context.setAllowUnmappedFields(true);
```

Comment:
The fieldnames are random because I mainly want to test the ability of the query parser to deal with varying field names and to check that serialization works. These tests are no integration tests, so I was trying to avoid testing different paths for whether fields have mappings or not. The problem with testing the constructed Lucene queries is that all depends very much on the implementation and testing which check the correct behaviour for mapped/unmapped fields should better be single node / integration tests.